src/datatype/svecf32.rs

```diff
@@ -0,0 +1,770 @@
+    fn argument_sql() -> Result<SqlMapping, ArgumentError> {
+        Ok(SqlMapping::As(String::from("svector")))
+    }
+    fn return_sql() -> Result<Returns, ReturnsError> {
+        Ok(Returns::One(SqlMapping::As(String::from("svector"))))
+    }
+}
+
+#[pgrx::pg_extern(immutable, parallel_safe, strict)]
+fn _vectors_svecf32_in(input: &CStr, _oid: Oid, _typmod: i32) -> SVecf32Output {
+    fn solve<T>(option: Option<T>, hint: &str) -> T {
+        if let Some(x) = option {
+            x
+        } else {
+            SessionError::BadLiteral {
+                hint: hint.to_string(),
+            }
+            .friendly()
+        }
+    }
+    #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+    enum State {
+        MatchingLeft,
+        Reading,
+        MatchedRight,
+    }
+    use State::*;
+    let input = input.to_bytes();
+    let mut indexes = Vec::<u16>::new();
+    let mut values = Vec::<F32>::new();
+    let mut state = MatchingLeft;
+    let mut token: Option<String> = None;
+    let mut index = 0;
+    for &c in input {
+        match (state, c) {
+            (MatchingLeft, b'[') => {
+                state = Reading;
+            }
+            (Reading, b'0'..=b'9' | b'a'..=b'z' | b'A'..=b'Z' | b'.' | b'+' | b'-') => {
+                let token = token.get_or_insert(String::new());
+                token.push(char::from_u32(c as u32).unwrap());
+            }
+            (Reading, b',') => {
+                let token = solve(token.take(), "Expect a number.");
+                let value: F32 = solve(token.parse().ok(), "Bad number.");
+                if !value.is_zero() {
+                    indexes.push(index);
+                    values.push(value);
+                }
+                index = match index.checked_add(1) {
+                    Some(x) => x,
+                    None => SessionError::BadValueDimensions.friendly(),
+                };
+            }
+            (Reading, b']') => {
+                if let Some(token) = token.take() {
+                    let value: F32 = solve(token.parse().ok(), "Bad number.");
+                    if !value.is_zero() {
+                        indexes.push(index);
+                        values.push(value);
+                    }
+                    index = match index.checked_add(1) {
+                        Some(x) => x,
+                        None => SessionError::BadValueDimensions.friendly(),
+                    };
+                }
+                state = MatchedRight;
+            }
+            (_, b' ') => {}
+            _ => {
+                SessionError::BadLiteral {
+                    hint: format!("Bad character with ascii {:#x}.", c),
+                }
+                .friendly();
+            }
+        }
+    }
+    if state != MatchedRight {
+        SessionError::BadLiteral {
+            hint: "Bad sequence.".to_string(),
+        }
+        .friendly();
+    }
+    SVecf32::new_in_postgres(SparseF32Ref {
```

Comment:
We need to check if the dimensions are in `[1, 65535]`.

Reply:
done in new commit